common/server_cardzone.cpp

```diff
@@ -135,7 +135,9 @@ int Server_CardZone::removeCard(Server_Card *card, bool &wasLookedAt)
     if (wasLookedAt && cardsBeingLookedAt > 0) {
         cardsBeingLookedAt -= 1;
     }
-    cards.removeAt(index);
+    if (index != -1) {
```

Comment:
why not start this function with if index == -1 return -1?

Reply:
It's possible the card might have coords, then we'd fail to remove it

Reply:
if the card has coords but is not present in this zone we have a different problem, actually just don't do this check at all, if this happens something went very wrong and we should stop it before it can do this

Reply:
Hmm, you make a good point. I'll change it back